### PR TITLE
implement project cards hover and modal

### LINKED ISSUE #14: project cards improvements
Right now, each project card in the Team and Projects sections only show a static image and a pretty bad placeholder title. They should:

* have good fallback behaviour for `bannerHasName: false` that looks good
* have on-hover interaction
* a simple modal on click to show the description and links (`links.repository`, `links.website`, etc.)

### PR DESCRIPTION
**Pull request by RachitMalik12**
**What's done:** 
- Modal code on click of thumbnail 
- Wired up data to modal
- Hover effect of cards and edge case where banner doesn't have the name.  

**Screenshots of what's done** 
Transparent:

Screen Shot 2020-04-22 at 12 50 42 PM

Background:

Screen Shot 2020-04-22 at 4 45 17 PM

Animations:

Screen-Recording-2020-04-22-at-6 53 40-PM

**What's left:** 
-  <del> Animation of cards and edge case where banner doesn't have the name </del>
- <del> Align the github logo and get better assets that go well with the background </del>

### CODE REVIEW

File: public/index.html
Comment (bobheadxi) [inline public/index.html]: What does this do? I took a quick look and it seems like `bulma-modal-fx`'s effects can be achieved with [`animate.css`](https://daneden.github.io/animate.css/), which we already depend on - our third-party CSS bundle is already quite large, let's try to avoid adding additional dependencies unless absolutely necessary

File: src/assets/icons/github.svg
Comment (bobheadxi) [inline src/assets/icons/github.svg:1]: this doesn't quite seem like the GitHub icon from the old side, which I think might look a bit better

**Review by bobheadxi — COMMENTED** [review]
looks like a great start! 🚀 just a thought: if we go with the transparent background modal, should we just make it full-screen so that the closing `x` isn't floating in what seems like the middle of the screen?

File: public/index.html
Comment (RachitMalik12) [reply public/index.html]: Yeah I'll remove it. I am not actually using it, I added it when I was playing around with stuff

File: src/assets/icons/github.svg
Comment (bobheadxi) [reply src/assets/icons/github.svg:1]: I wonder if we should just use `fontawesome` - see #46

Comment (RachitMalik12) [reply src/assets/icons/github.svg:1]: Let's use fontawesome

**Comment by RachitMalik12** [comment]
> Are the modals going to be have no non-transparent background? I think the modals should have a solid background and the site in the background can have the opacity reduced for readability.

Yeah that works and the original intention was a solid background for the modal. I was playing around and by default bulma does modals as transparent. Now that I look at it it doesn't look too shabby. I know in general most modals have a solid background but in this case I think the text readability isn't affected. However, I can try one with the background and see whether it works better. Here it is:

Screen Shot 2020-04-22 at 4 45 17 PM

**Comment by bobheadxi** [comment]
I like the hover behaviour for images without a title! Thought: a gentler glow/light drop shadow would probably look better on hover

image

The website is dark, but something reminiscent of one of @cowjuh 's showcase designs would be nice imo:

image

**Comment by RachitMalik12** [comment]
> I like the hover behaviour for images without a title! Thought: a gentler glow/light drop shadow would probably look better on hover
> 
How does this look? 


Screen Shot 2020-04-22 at 9 31 22 PM

**Comment by bobheadxi** [comment]
@RachitMalik12 looks better! maybe add a little more spread to the glow?

**Comment by srijonsaha** [comment]
@RachitMalik12 try removing the border on hover and settings `box-shadow: 0px 0px 15px 8px rgba($white, 0.35);` to make it pop out more

**Comment by bobheadxi** [comment]
> @RachitMalik12 try removing the border on hover and setting box-shadow: 0px 0px 15px 8px rgba($white, 0.35); to make it pop out more

Sorry for the nit, but still unsure about how 5bd8cad looks 😅 I think we should go for an even spread around the tile since it's not a real "shadow" and should be more of a uniform glow

image

File: src/components/Modal.vue
Comment (bobheadxi) [inline src/components/Modal.vue]:
```suggestion
```

ID seems unused

Comment (bobheadxi) [reply src/components/Modal.vue:3]: Try adding this:

```suggestion
    class="animated zoomIn"
```

File: src/components/Modal.vue
Comment (bobheadxi) [inline src/components/Modal.vue]: try this:

```suggestion
      'modal is-active animated zoomIn': isActive,
```

File: src/components/Modal.vue
Comment (bobheadxi) [inline src/components/Modal.vue]: to align with the name of the card component, maybe something more descriptive like:

```suggestion
  name: 'TeamProjectModal',
```

and similar change for the filename?

File: src/components/Modal.vue
Comment (bobheadxi) [inline src/components/Modal.vue]: `website` and `writeup` are [optional](https://ubclaunchpad.github.io/new/config/globals.html#project) - we can get rid of the placeholders and just not render these divs using a `v-if="team.project.links.website"` directive

Comment (RachitMalik12) [reply src/components/Modal.vue]: Yep currently doing that

Comment (RachitMalik12) [reply src/components/Modal.vue]: I was thinking of also adding iOS and Android config options and icons. Currently in progress..

Comment (bobheadxi) [reply src/components/Modal.vue]: > iOS and Android config options and icons

hold on to that thought, maybe open another ticket - let's keep this PR on-scope for the time being :) most notably I think it should wait until post- #46

File: package.json
Comment (bobheadxi) [inline package.json]:
```suggestion
```

File: public/index.html
Comment (bobheadxi) [inline public/index.html:11]: Is this meant to resolve #46 ?

File: src/components/Modal.vue
Comment (RachitMalik12) [reply src/components/Modal.vue]: Yeah.. Which is why I removed the iconset stuff as well. I'll do it in another PR and make an issue for the config.

File: public/index.html
Comment (RachitMalik12) [reply public/index.html:11]: I was thinking of resolving #46 in another PR to keep this one on scope. We might have a dangling font awesome lying there but I will make the next PR immediately after this one goes in.

Comment (bobheadxi) [reply public/index.html:11]: Sounds good! 🎉

File: src/components/TeamProjectCard.vue
Comment (srijonsaha) [inline src/components/TeamProjectCard.vue:62]: There should be 4 px units. x offset, y offset, blur, spread. The 15px was supposed to set the blur but right settings the y offset which doesn't look that great since we want an even shadow effect.

Comment (RachitMalik12) [reply src/components/TeamProjectCard.vue]: Yep my bad 😅 .  looks like this now so more even: 
Screen Shot 2020-04-23 at 11 21 00 AM

File: src/components/Modal.vue
Comment (RachitMalik12) [reply src/components/Modal.vue]: Another thing that I thought of was to make the Modal more generic so we can use the same component in different scenarios. However that would need some code refactoring. For now I am open to making this a more specific component in the interest of time so I'll change the name.

File: src/components/Modal.vue
Comment (RachitMalik12) [reply src/components/Modal.vue]: So actually I think it's better if we do that to the modal content since doing that to the main modal div animates the whole blurred background around the content. Looks better when we do it that way.

File: package.json
Comment (RachitMalik12) [reply package.json]: Fixed in: c525c21

File: src/components/Modal.vue
Comment (RachitMalik12) [reply src/components/Modal.vue]: Fixed in: c525c21

File: src/components/Modal.vue
Comment (RachitMalik12) [reply src/components/Modal.vue]: Fixed in: c525c21

File: src/components/Modal.vue
Comment (RachitMalik12) [reply src/components/Modal.vue]: Done ^ in c525c21

**Review by bobheadxi — APPROVED** [review]
imo this is a good start! I think we can make additional improvements down the line.

I just have *one last nit*: it seems in the on-hover for projects to show names, the name itself also gets the transparency applied, when it should be completely opaque:

image

if that proves difficult to fix, I don't mind holding off on it for now. let's get an approve from @srijonsaha as well before we merge?

**Comment by RachitMalik12** [comment]
> imo this is a good start! I think we can make additional improvements down the line.
> 
> I just have _one last nit_: it seems in the on-hover for projects to show names, the name itself also gets the transparency applied, when it should be completely opaque:
> 
> image

Yeah it bugs me too. I'll try to fix it.

> if that proves difficult to fix, I don't mind holding off on it for now. let's get an approve from @srijonsaha as well before we merge?

Sure! Thanks!

**Comment by bobheadxi** [comment]
@RachitMalik12  just merged #61 - let's see how the new config looks with this branch?

**Comment by RachitMalik12** [comment]
> imo this is a good start! I think we can make additional improvements down the line.
> 
> I just have _one last nit_: it seems in the on-hover for projects to show names, the name itself also gets the transparency applied, when it should be completely opaque:
> 
> image
@bobheadxi 
**Looks like this now**, a little better: 

Screen Shot 2020-04-23 at 10 59 23 PM

**Comment by bobheadxi** [comment]
> Looks like this now, a little better

A lot better, thank you!

**Pull request merged**
